### PR TITLE
Enable package visibility feature for tools-version 5.9

### PR DESCRIPTION
**Pull request by elsh**
The package visibility proposal has officially been accepted: https://forums.swift.org/t/accepted-se-0386-package-access-modifier/64904
Resolves rdar://109152010

### CODE REVIEW

File: Sources/PackageDescription/Target.swift [hunk 1]
```diff
@@ -568,7 +568,7 @@ public final class Target {
     ///   - swiftSettings: The Swift settings for this target.
     ///   - linkerSettings: The linker settings for this target.
     ///   - plugins: The plug-ins used by this target
-    @available(_PackageDescription, introduced: 999.0)
+  @available(_PackageDescription, introduced: 5.9)
```
Comment (MaxDesiatov) [inline Sources/PackageDescription/Target.swift:571]: nit: formatting seems off

**Review by MaxDesiatov — APPROVED** [review]

**Comment by elsh** [comment]
@swift-ci smoke test

**Comment by elsh** [comment]
@swift-ci smoke test

File: Tests/BuildTests/BuildPlanTests.swift [hunk 2]
```diff
@@ -552,7 +552,7 @@ final class BuildPlanTests: XCTestCase {
             if isFlagSupportedInDriver {
                 XCTAssertMatch(stdout, .contains("-package-name apppkg"))
                 XCTAssertMatch(stdout, .contains("-package-name foopkg"))
-                // the flag is not supported in old toolsversions
+                // the flag is not supported if < 5.9
```
Comment (neonichu) [inline Tests/BuildTests/BuildPlanTests.swift:555]:
```suggestion
                // the flag is not supported if tools-version < 5.9
```

**Comment by neonichu** [comment]
Should we add the flag to `Target.macro` as well for completeness?

**Comment by elsh** [comment]
> Should we add the flag to `Target.macro` as well for completeness?

Looks like it was added for tools version 5.9 already in main; https://github.com/apple/swift-package-manager/blob/main/Sources/CompilerPluginSupport/TargetExtensions.swift#L23

**Comment by elsh** [comment]
@swift-ci smoke test

**Comment by elsh** [comment]
@swift-ci smoke test windows

**Comment by elsh** [comment]
@swift-ci smoke test windows

**Comment by elsh** [comment]
@swift-ci smoke test windows

**Comment by elsh** [comment]
@swift-ci smoke test windows

**Pull request merged**